components/org.wso2.carbon.crypto.provider/src/main/java/org/wso2/carbon/crypto/provider/KeyStoreBasedInternalCryptoProvider.java

```diff
@@ -137,8 +152,9 @@ public KeyStoreBasedInternalCryptoProvider(KeyStore keyStore, String keyAlias, S
             return cipher.doFinal(ciphertext);
         } catch (NoSuchAlgorithmException | NoSuchPaddingException | NoSuchProviderException | BadPaddingException
                 | IllegalBlockSizeException | InvalidKeyException | UnrecoverableKeyException | KeyStoreException e) {
-            String errorMessage = String.format("An error occurred while decrypting using the algorithm : '%s'"
-                    , algorithm);
+            String errorMessage = String.format("An error occurred while decrypting using the algorithm : '%s', and " +
+                            "crypto provider : '%s'"
+                    , algorithm, this.getClass().getName());
```

Comment:
Shall we move this line up

Reply:
Fixed with 859521e2f1237359b0fd32027af5d18c51606b22